### ISSUE TITLE
Socket reads with SSL delegate recv failing on osx and linux

### ISSUE THREAD

**Issue by na-gupta** [IBM-Swift/BlueSSLService#22]
Socket reads with SSL delegate recv() fails on linux and mac depending on the size of the incoming data.

It looks like it has something to do with how errSSLWouldBlock/SSL_ERROR_WANT_* is handled in SSLService recv:
```
#if os(Linux)
    if lastError == SSL_ERROR_WANT_READ || lastError == SSL_ERROR_WANT_WRITE {
        errno = EAGAIN
        return 0
    }
#else
    if status == errSSLWouldBlock {
        errno = EWOULDBLOCK
        return Int(errSSLWouldBlock)
    }
#endif
```
... but in Socket `readDataIntoStorage()` and `readDatagramIntoStorage()`, it appears to be expecting `delegate.recv()` to `throw SSLError.retryNeeded` when this happens.

It appears to fails when the data is >= 4096 bytes and 
- on linux when the data is an exact multiple of 4096 (say 4096 bytes)
- on osx when the data is NOT an exact multiple of 4096 (say 4097 bytes)

**Comment by shmuelk** [comment]
Inside BlueSocket there is an internal buffer which I believe by default is 4K in size.

**Comment by na-gupta** [comment]
This issue can be replicated on linux and osx using the [BlueSSLService_22 branch](https://github.com/IBM-Swift/Kitura/tree/BlueSSLService_22) of Kitura by running `swift test -s KituraTests.TestResponse/testLargePost`

**Comment by na-gupta** [comment]
@shmuelk Yes, @billabt and I were chatting about that. That buffer's size is definitely part of the cause but the fix is unclear.

**Comment by billabt** [comment]
This is fixed in *v0.12.39 of BlueSocket*.
